resolwe_bio/processes/reads_processing/cutadapt.yml

```diff
@@ -25,93 +21,112 @@
-    - name: crop
-      label: Crop
-      type: basic:integer
-      required: false
-      description: |
-       Cut the read to a specified length by removing bases from the end
-    - name: headcrop
-      label: Headcrop
-      type: basic:integer
-      required: false
-      description: |
-       Cut the specified number of bases from the start of the read
-    - name: min_overlap
-      label: Minimal overlap
-      type: basic:integer
-      default: 3
-      description: |
-       Minimum overlap for an adapter match
-    - name: minlen
-      label: Min length
-      type: basic:integer
-      required: false
-      description: |
-        Drop the read if it is below a specified length
-    - name: max_n
-      label: Max numebr of N-s
-      type: basic:integer
-      required: false
-      description: |
-        Discard reads having more 'N' bases than specified.
-    - name: match_read_wildcards
-      label: Match read wildcards
-      type: basic:boolean
-      required: false
-      default: false
-      description: |
-        Interpret IUPAC wildcards in reads.
-    - name: times
-      label: Times
-      type: basic:integer
-      required: false
-      default: 1
-      description: |
-        Remove up to COUNT adapters from each read.
-    - name: error_rate
-      label: Error rate
-      type: basic:decimal
-      required: false
-      default: 0.1
-      description: |
-        Maximum allowed error rate (no. of errors divided by the length of the matching region).
+    - name: adapters
+      label: Adapters
+      group:
+        - name: up_primers_file
+          label: 5 prime adapter file
+          type: data:seq:nucleotide
+          required: false
+        - name: down_primers_file
+          label: 3 prime adapter file
+          type: data:seq:nucleotide
+          required: false
+        - name: up_primers_seq
+          label: 5 prime adapter sequence
+          type: list:basic:string
+          required: false
+        - name: down_primers_seq
+          label: 3 prime adapter sequence
+          type: list:basic:string
+          required: false
+        - name: polya_tail
+          label: Poly-A tail
+          type: basic:integer
+          required: false
+          description: |
+            Length of poly-A tail, example - AAAN -> 3, AAAAAN -> 5
+        - name: min_overlap
+          label: Minimal overlap
+          type: basic:integer
+          default: 3
+          description: |
+            Minimum overlap for an adapter match
+    - name: modify_reads
+      label: Read trimming
+      group:
+        - name: nextseq_trim
+          label: NextSeq-specific quality trimming
+          type: basic:integer
+          required: false
+          description: |
+            NextSeq-specific quality trimming (each read). Trims also dark
+            cycles appearing as high-quality G bases. This option is mutually
```

Comment:
In one of the future versions we should make a drop-down menu for selection of quality trimming type. This would make mutual exclusion mentioned in the description more obvious.